lib/Dialect/FIRRTL/Transforms/PrefixModules.cpp

```diff
@@ -304,38 +304,46 @@ void PrefixModulesPass::renameModule(FModuleOp module) {
                   StringAttr::get(module.getContext(), newModuleName));
   renameModuleBody(prefixFull, module);
 
-  // If this module contains a Grand Central interface, then also apply renames
-  // to that, but only if there are prefixes to apply.
   AnnotationSet annotations(module);
-  if (!annotations.hasAnnotation(parentAnnoClass) &&
-      !annotations.hasAnnotation(companionAnnoClass))
-    return;
-  SmallVector<Attribute> newAnnotations;
-  for (auto anno : annotations) {
-    if (!anno.isClass(parentAnnoClass) && !anno.isClass(companionAnnoClass)) {
+  SmallVector<Attribute, 1> newAnnotations;
+  annotations.removeAnnotations([&](Annotation anno) {
+    if (anno.getClass() == dutAnnoClass) {
+      anno.setMember("prefix", builder.getStringAttr(prefixFull));
       newAnnotations.push_back(anno.getDict());
-      continue;
+      return true;
     }
 
-    NamedAttrList newAnno;
-    for (auto attr : anno) {
-      if (attr.getName() == "name") {
-        newAnno.append(attr.getName(),
-                       (Attribute)builder.getStringAttr(
-                           Twine(prefixFull) +
-                           attr.getValue().cast<StringAttr>().getValue()));
-        continue;
+    bool remove = false;
```

Comment:
Consider a more descriptive name for this.

Reply:
I think I can get rid of the need for it actually by pulling lines 337-8 into the if starting on line 318, let me tweak it